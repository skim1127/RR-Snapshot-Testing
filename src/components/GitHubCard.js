import Card from 'react-bootstrap/Card'

export default function GitHubCard() {
    return(
        <Card>
            <Card.Img variant="top" src="../images/doge.png"/>
            <Card.Body>
                <Card.Title>skim1127</Card.Title>
                <Card.Text>
                    Upcoming Full-Stack Developer :)
                </Card.Text>
            </Card.Body>
        </Card>
    )
}