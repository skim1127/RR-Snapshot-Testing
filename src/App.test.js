import { render, screen } from '@testing-library/react';
import App from './App';
import renderer from 'react-test-renderer'
import GitHubCard from './components/GitHubCard';

// Snapshot Testing
test('renders a snapshot', () => {
  // The call toJSON is important 
  // because it turns the component representation into JSON
  const tree = renderer.create(<App/>).toJSON()
  // console.log(tree)
  expect(tree).toMatchSnapshot()
})

// Snapshot testing for GitHubCard Component
test('renders a github card component', () => {
  const tree = renderer.create(<GitHubCard/>).toJSON()
  console.log(tree)
  expect(tree).toMatchSnapshot()
})