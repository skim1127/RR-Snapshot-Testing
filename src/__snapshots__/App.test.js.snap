// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a github card component 1`] = `
<div
  className="card"
>
  <img
    className="card-img-top"
    src="../images/doge.png"
  />
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      skim1127
    </div>
    <p
      className="card-text"
    >
      Upcoming Full-Stack Developer :)
    </p>
  </div>
</div>
`;

exports[`renders a snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
  <h2>
    Hello World
  </h2>
</div>
`;
